Open Judge System/Tools/ExecutionStrategiesMigrator/Startup.cs

```diff
@@ -0,0 +1,120 @@
+﻿namespace ExecutionStrategiesMigrator
+{
+    using System;
+    using System.Data.Entity;
+    using System.Linq;
+
+    using OJS.Data;
+    using OJS.Data.Models;
+
+    public class Startup
+    {
+        public static void Main()
+        {
+            var db = GetData();
+
+            var processed = 0;
+            var page = 0;
+            var pageSize = 500;
+
+            while (true)
+            {
+                Console.WriteLine("Reading data for contests...");
+
+                var allProblems = GetAllProblems(db.Problems, page, pageSize);
+                if (!allProblems.Any())
+                {
+                    break;
+                }
+
+                page++;
+
+                foreach (var problem in allProblems)
+                {
+                    problem.SubmissionTypes = problem.Contest.SubmissionTypes;
+                    db.Entry(problem).State = EntityState.Modified;
+                    Console.WriteLine($"Problem with id = {problem.Id} modified");
+                    processed++;
+
+                    if (processed % 500 == 0)
+                    {
+                        db.SaveChanges();
+                        db = GetData();
+                    }
+
+                    if (processed % 500 == 0)
+                    {
+                        Console.WriteLine($"{processed} problems processed");
+                    }
+
+                }
+            }
+
+            db.SaveChanges();
+            db = GetData();
+
+            Console.WriteLine("All submissions processed");
+
+            processed = 0;
+            page = 0;
+            pageSize = 500;
+
+            while (true)
```

Comment:
I'm not sure that this has some point at all. You can test it when you write it, but what's the point of that if you know that the code is right?